contrib/platform/test/com/sun/jna/platform/win32/OleaccTest.java

```diff
@@ -0,0 +1,152 @@
+        PointerByReference pointer = new PointerByReference();
+        HRESULT hresult = Oleacc.INSTANCE.AccessibleObjectFromWindow(hwnd, 0, riid, pointer);
+        assertEquals(S_OK, hresult);
+        return new Accessible(pointer.getPointer().getPointer(0L));
+    }
+
+    @Test
+    public void testAccessibleChildren() {
+        // Get accessible from hwnd first
+        Accessible accessible = getCalculatorAccessible();
+
+        // Call AccessibleChildren
+        IntByReference cChildren = new IntByReference();
+        HRESULT hresult1 = accessible.get_accChildCount(cChildren);
+        assertEquals(S_OK, hresult1);
+
+        Variant.VARIANT[] rgvarChildren = new Variant.VARIANT[(int) cChildren.getValue()];
+        IntByReference pcObtained = new IntByReference();
+
+        HRESULT hresult2 = Oleacc.INSTANCE.AccessibleChildren(
+                accessible.getPointer(),
+                0,
+                cChildren.getValue(),
+                rgvarChildren,
+                pcObtained
+        );
+        assertEquals(S_OK, hresult2);
+        assertEquals(7, pcObtained.getValue());
+
+        for (int i = 0; i < pcObtained.getValue(); ++i) {
+            assertEquals(Variant.VT_DISPATCH, rgvarChildren[i].getVarType().intValue());
+        }
+    }
+
+    @Test
+    public void testAccessibleObjectFromWindow() {
+        HWND hwnd = getCalculatorHwnd();
+        REFIID riid = new REFIID(IAccessible.IID_IACCESSIBLE);
+        PointerByReference pointer = new PointerByReference();
+        HRESULT hresult = Oleacc.INSTANCE.AccessibleObjectFromWindow(hwnd, 0, riid, pointer);
+        assertEquals(S_OK, hresult);
+    }
+
+    @Test
+    public void testWindowFromAccessibleObject() {
+        // Get accessible from hwnd first
+        HWND originalHwnd = getCalculatorHwnd();
+        Accessible accessible = getCalculatorAccessible();
+
+        // Then attempt to get same hwnd from accessible
+        PointerByReference phwnd = new PointerByReference();
+        HRESULT hresult2 = Oleacc.INSTANCE.WindowFromAccessibleObject(accessible.getPointer(), phwnd);
+        assertEquals(S_OK, hresult2);
+        assertEquals(originalHwnd, new HWND(phwnd.getPointer().getPointer(0L)));
+    }
+
+    @Test
+    public void testGetRoleTextA() {
+        int result = Oleacc.INSTANCE.GetRoleTextA(Oleacc.ROLE_SYSTEM_TITLEBAR, null, 0);
+        assertEquals(9, result);
+        LPSTR lptstr = new LPSTR(new Memory(result + 1)); // plus 1 for null terminator
+        int result2 = Oleacc.INSTANCE.GetRoleTextA(Oleacc.ROLE_SYSTEM_TITLEBAR, lptstr, result + 1);
+        assertEquals(result, result2);
+        assertEquals("title bar", lptstr.toString());
+    }
+
+    @Test
+    public void testGetRoleTextW() {
+        int result = Oleacc.INSTANCE.GetRoleTextW(Oleacc.ROLE_SYSTEM_TITLEBAR, null, 0);
+        assertEquals(9, result);
+        LPWSTR lpwstr = new LPWSTR(new Memory(result + 1)); // plus 1 for null terminator
```

Comment:
This is wrong and will result in an out of bounds write. `GetRoleTextW` returns the number of unicode characters, not bytes. So this should be:

```
LPWSTR lpwstr = new LPWSTR(new Memory((result + 1) * Native.WCHAR_SIZE));
```

I suggest to not bind `GetRoleTextA` and `GetRoleTextW` indiviually, but only once as:

```
int GetRoleText(int lRole, Pointer lpszRole, int cchRoleMax);
```

And add a

```
String GetRoleText(int iRole)
```

in `OleAccUtil`. That helper can than use the correct variant based on the value of `w32.ascii`. This is a sample:

https://github.com/java-native-access/jna/blob/7dbbf4b447674182e5752ac9b1eba91d9bffc22e/contrib/platform/src/com/sun/jna/platform/win32/Crypt32Util.java#L221-L249

Reply:
Yeah great idea, that works out to be much nicer to write and use too. I've made this change and added a `OleAccUtil` file.